spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -1473,9 +1503,371 @@ static void queryWithQueryOptions(DatabaseClient dbClient) {
+    if (!backup.isReady()) {
+      System.out.println("Backup is not ready for use.");
+      throw new RuntimeException("Backup is not ready for use.");
+    }
+    // Reload the metadata of the backup from the server.
+    backup = backup.reload();
+    System.out.println(
+        String.format(
+            "Backup %s of size %d bytes was created at %s",
+            backup.getId().getName(),
+            backup.getSize(),
+            LocalDateTime.ofEpochSecond(
+                backup.getProto().getCreateTime().getSeconds(),
+                backup.getProto().getCreateTime().getNanos(),
+                OffsetDateTime.now().getOffset())).toString());
+  }
+  // [END spanner_create_backup]
+
+  // [START spanner_cancel_create_backup]
+  static void cancelCreateBackup(
+      DatabaseAdminClient dbAdminClient, DatabaseId databaseId) {
+    String backupName =
+        String.format(
+            "%s_%02d_cancel",
+            databaseId.getDatabase(), LocalDate.now().get(ChronoField.ALIGNED_WEEK_OF_YEAR));
+    BackupId backupId = BackupId.of(databaseId.getInstanceId(), backupName);
+    // Set expire time to 14 days from now.
+    long currentTimeInMicroSeconds =
+        TimeUnit.MICROSECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS);
+    long deltaTimeInMicroseconds = TimeUnit.MICROSECONDS.convert(14L, TimeUnit.DAYS);
+    Timestamp expireTime =
+        Timestamp.ofTimeMicroseconds(currentTimeInMicroSeconds + deltaTimeInMicroseconds);
+
+    // Create a backup instance.
+    Backup backup =
+        dbAdminClient
+            .newBackupBuilder(backupId)
+            .setDatabase(databaseId)
+            .setExpireTime(expireTime)
+            .build();
+    // Start the creation of a backup.
+    System.out.println("Creating backup [" + backup.getId() + "]...");
+    OperationFuture<Backup, CreateBackupMetadata> op = backup.create();
+    try {
+      // Try to cancel the backup operation.
+      System.out.println("Cancelling create backup operation for [" + backup.getId() + "]...");
+      dbAdminClient.cancelOperation(op.getName());
+      // Get a polling future for the running operation. This future will regularly poll the server
+      // for the current status of the backup operation.
+      RetryingFuture<OperationSnapshot> pollingFuture = op.getPollingFuture();
+      // Wait for the operation to finish.
+      // isDone will return true if the operation has finished successfully or if it was cancelled
+      // or any other error occurred..
+      while (!pollingFuture.get().isDone()) {
+        System.out.println("Waiting for the cancelled backup operation to finish...");
+        Thread.sleep(TimeUnit.MILLISECONDS.convert(5, TimeUnit.SECONDS));
+      }
+      if (pollingFuture.get().getErrorCode() == null) {
+        // Backup was created before it could be cancelled. Delete the backup.
+        System.out.println("Deleting backup [" + backup.getId() + "]...");
```

Comment:
I think the comment is sufficient and we can get rid of the println.

Reply:
done